pkg/sql/copy_file_upload.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/blobs"
+	"github.com/cockroachdb/cockroach/pkg/internal/client"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirebase"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/lib/pq"
+)
+
+const (
+	fileUploadTable = "crdb_internal.file_upload"
+	copyOptionDest  = "destination"
+)
+
+var copyFileOptionExpectValues = map[string]KVStringOptValidate{
+	copyOptionDest: KVStringOptRequireValue,
+}
+
+type copyError struct {
```

Comment:
why does this type need to exist?

Reply:
No, it doesn't I had some confusion about the behaviour of goroutines, removing this.